apps/dashboard/app/models/posix_file.rb

```diff
@@ -202,6 +202,17 @@ def can_download_as_zip?(timeout: Configuration.file_download_dir_timeout, downl
     return can_download, error
   end
 
+  # This serves the same function as can_download_as_zip?, but for files
+  def can_download_file?(download_file_size_limit: Configuration.file_download_max)
+    if !(file? && readable?)
+      error = I18n.t('dashboard.files_directory_download_unauthorized')
+      return [false, error]
+    end
```

Comment:
Do we not already do this somewhere else?

Also seems like we should use `unless` instead of `if !`